src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -321,7 +331,7 @@ export class StudyViewPageStore {
     }
 
     @action
-    updateStoreFromURL(query: Partial<StudyViewURLQuery>) {
+    async updateStoreFromURL(query: Partial<StudyViewURLQuery>) {
```

Comment:
async actions need special treatment.   also, there is no way to reflect the loading state of this behavior in the UI.

Reply:
also, we need tests for this stuff.  lot of logic in here.  it's pretty important because it determines the results the user sees.